src/back/glsl/mod.rs

```diff
@@ -1487,6 +1492,41 @@ impl<'a, W: Write> Writer<'a, W> {
         Ok(())
     }
 
+    fn write_workgroup_variables_initialization(
+        &mut self,
+        ctx: &back::FunctionCtx,
+    ) -> BackendResult {
+        if !self.options.zero_initialize_workgroup_memory
+            || !ctx.ty.is_compute_entry_point(self.module)
+        {
+            return Ok(());
+        }
+
+        let vars = self
+            .module
+            .global_variables
+            .iter()
+            .filter(|&(handle, var)| {
+                !ctx.info[handle].is_empty() && var.space == crate::AddressSpace::WorkGroup
+            });
+
+        let mut write_barrier = false;
+
+        for (handle, var) in vars {
+            let name = &self.names[&NameKey::GlobalVariable(handle)];
+            write!(self.out, "{}{} = ", back::INDENT, name)?;
+            self.write_zero_init_value(var.ty)?;
+            writeln!(self.out, ";")?;
+            write_barrier = true;
+        }
+
+        if write_barrier {
+            self.write_barrier(crate::Barrier::WORK_GROUP, back::Level(1))?;
```

Comment:
question: Where can I learn more about this?

TODO: try to find out about this myself first.